src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GCNVPostProcessor.java

```diff
@@ -0,0 +1,197 @@
+        variantContextBuilder.alleles(alleles);
+        variantContextBuilder.chr(copyNumberPosteriorLocatableRecord.getContig());
+        variantContextBuilder.start(copyNumberPosteriorLocatableRecord.getStart());
+        variantContextBuilder.stop(copyNumberPosteriorLocatableRecord.getEnd());
+        variantContextBuilder.id(String.format(variantPrefix + "_%s_%d_%d",
+                copyNumberPosteriorLocatableRecord.getContig(),
+                copyNumberPosteriorLocatableRecord.getStart(),
+                copyNumberPosteriorLocatableRecord.getEnd()));
+        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(sampleName);
+        final List<Integer> copyNumberPLVector = getCopyNumberPLVector(copyNumberPosteriorLocatableRecord, integerCopyNumberStateCollection);
+        final int copyNumberMAP = calculateMAPCopyNumberState(copyNumberPosteriorLocatableRecord, integerCopyNumberStateCollection);
+        final int GQ = calculateGenotypeQuality(copyNumberPosteriorLocatableRecord, integerCopyNumberStateCollection);
+        genotypeBuilder.attribute(CN, copyNumberMAP);
+        genotypeBuilder.attribute(CNLP, copyNumberPLVector);
+        genotypeBuilder.attribute(CNQ, GQ);
+
+        final Genotype genotype = genotypeBuilder.make();
+
+        //Add allele information to the variant context
+        variantContextBuilder.attribute(VCFConstants.END_KEY, copyNumberPosteriorLocatableRecord.getEnd());
+
+        variantContextBuilder.genotypes(genotype);
+        return variantContextBuilder.make();
+    }
+
+    @VisibleForTesting
+    static List<Integer> getCopyNumberPLVector(final CopyNumberPosteriorLocatableRecord copyNumberPosteriorLocatableRecord,
+                                                     final IntegerCopyNumberStateCollection integerCopyNumberStateCollection) {
+        final List<Integer> unnormalizedPL = integerCopyNumberStateCollection.getCopyNumberStates().stream()
+                .mapToDouble(state -> copyNumberPosteriorLocatableRecord.getCopyNumberPosteriors().getCopyNumberPosterior(state))
+                .mapToInt(GCNVPostProcessor::convertLogPosteriorToPL).boxed().collect(Collectors.toList());
+        final Optional<Integer> smallestPL = unnormalizedPL.stream().min(Integer::compare);
+        return unnormalizedPL.stream().mapToInt(p -> p - smallestPL.get()).boxed().collect(Collectors.toList());
+    }
+
+    @VisibleForTesting
+    static int calculateMAPCopyNumberState(final CopyNumberPosteriorLocatableRecord copyNumberPosteriorLocatableRecord,
+                                      final IntegerCopyNumberStateCollection integerCopyNumberStateCollection) {
+        final Optional<IntegerCopyNumberState> copyNumberStateMAP = integerCopyNumberStateCollection.getCopyNumberStates()
+                .stream().max((a1, a2) -> Double.compare(copyNumberPosteriorLocatableRecord.getCopyNumberPosteriors().getCopyNumberPosterior(a1),
```

Comment:
intellij tells me this can be replaced by `Comparator.comparingDouble`

Reply:
Done